scripts/bad_words.py

```diff
@@ -0,0 +1,101 @@
+
+
+def main(data_api_url, data_api_token, bad_words_path, framework_slug, output_dir):
+    client = DataAPIClient(data_api_url, data_api_token)
+    bad_words = get_bad_words(bad_words_path)
+    suppliers = get_suppliers(client, framework_slug)
+    output_services_with_bad_words(output_dir, framework_slug, client, suppliers, bad_words)
+
+
+def get_suppliers(client, framework_slug):
+    suppliers = client.find_framework_suppliers(framework_slug)
+    suppliers = suppliers["supplierFrameworks"]
+    if (framework_slug == "g-cloud-6"):
+        suppliers_on_framework = suppliers
+    else:
+        suppliers_on_framework = [supplier for supplier in suppliers if supplier["onFramework"]]
+    return suppliers_on_framework
+
+
+def get_draft_services(client, supplier_id, framework_slug):
+    services = client.find_draft_services(supplier_id, framework=framework_slug)
+    services = services["services"]
+    submitted_services = [service for service in services if service["status"] == "submitted"]
+    return submitted_services
+
+
+def get_services(client, supplier_id, framework_slug):
+    services = client.find_services(supplier_id, framework=framework_slug)
+    services = services["services"]
+    return services
+
+
+def get_bad_words(bad_words_path):
+    with open(bad_words_path) as file:
+        lines = file.readlines()
+        return [line.strip() for line in lines if not (line.startswith("#") or line.isspace())]
+
+
+def output_services_with_bad_words(output_dir, framework_slug, client, suppliers, bad_words):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+    with open('{}/{}-services-with-blacklisted-words.csv'.format(
+            output_dir, framework_slug), 'w') as csvfile:
+        fieldnames = [
+            'Supplier ID',
+            'Supplier',
+            'Service ID',
+            'Service Title',
+            'Service Description',
+            'Blacklisted Word',
+            'Service Features',
+        ]
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames, dialect='excel')
+        writer.writeheader()
+        for supplier in suppliers:
+            if supplier["frameworkSlug"] == "g-cloud-6":
+                services = get_services(client, supplier["supplierId"], supplier["frameworkSlug"])
+            else:
+                services = get_draft_services(client, supplier["supplierId"], supplier["frameworkSlug"])
+            for service in services:
+                for key in service:
+                    if isinstance(service[key], six.string_types):
+                        for word in bad_words:
+                            if re.search(r"\b{}\b".format(word), service[key], re.IGNORECASE):
+                                service_features = ""
+                                for feature in service["serviceFeatures"]:
+                                    service_features += feature + ","
+                                service_features = service_features[:-1]
```

Comment:
Would it be better to flatten the whole service JSON and check that, or loop through each key and log that?

@carolinegreen `serviceFeatures` is one of a few keys that contain user-submitted content. Sorry I didn’t make that clear in my last comment.